src/openforms/forms/api/validators.py

```diff
@@ -30,8 +30,8 @@ def __call__(self, value: Union[dict, str]) -> None:
 
         expression = JSONLogicExpression.from_expression(value)
         tree = expression.as_tree()
-        assert isinstance(tree, Operation)
-        if tree.arguments[0] == "":
+        assert isinstance(tree, Operation | float | list)
+        if isinstance(tree, Operation) and tree.arguments[0] == "":
```

Comment:
the float should really go above with the other primitives, and you could re-use the type hint for this:

```py
from openforms.typing import JSONPrimitive

if isinstance(value, JSONPrimitive):
    return

super().__call__(value)

expression = SONLogicExpression.from_expression(value)
tree = expression.as_tree()
```
I'm trying to figure out where this validation comes from, from the commits it seems that #1472 is the reason:

> Validate that component can't be empty (both in the trigger and the actions) 

via #1499

But it seems that `test_cant_have_empty_component_value_in_value_action` is also a wrong promise, since `{"var": ""}` is a valid JSON logic expression referring to the identity of what you put in:

```py
>>> from json_logic import jsonLogic
>>> foo = {"bar": "baz"}
>>> jsonLogic({"var": ""}, foo)
{'bar': 'baz'}
```
So perhaps this validator doesn't even have a reason anymore to exist :thinking: And if it does exist, we should validate that the `Var` leaf nodes point to valid form variables, but that can in turn cause dead references (e.g. `{"var": {"var": "foo"}}` that can't be validated due to the dynamic nature. Probably those complex scenarios should be caught via static analysis for data types (big topic #2324 ).

So I'm leaning towards getting rid of this validator alltogether.

Reply:
@pi-sigma let's replace this validator with the base `JSONLogicValidator` and remove the associated test(s). It's no longer relevant.